test/integration/testEasyLoggingOnConnecting.js

```diff
@@ -0,0 +1,126 @@
+let tempDir = null;
+
+describe('Easy logging tests', function () {
+
+  before(async function () {
+    tempDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), 'easy_logging_connect_tests_'));
+  });
+
+  after(async function () {
+    Logger.getInstance().configure({
+      level: logLevelBefore,
+      filePath: 'snowflake.log'
+    });
+    await fsPromises.rm(tempDir, { recursive: true, force: true });
+  });
+
+  afterEach(function () {
+    resetEasyLoggingModule();
+  });
+
+  it('Should apply easy logging config when connection is being opened with callback', function (done) {
+    const logLevel = 'ERROR';
+    createConfigFile(logLevel).then((configFilePath) => {
+      const configParameters = createConfigParameters(configFilePath);
+      const connection = snowflake.createConnection(configParameters);
+      connection.connect((err) => {
+        if (err) {
+          done(err);
+        } else {
+          assert.strictEqual(Logger.getInstance().getLevelTag(), logLevel);
+          done();
+        }
+      });
+    });
+  });
+
+  it('Should fail for connecting with wrong easy logging config', function (done) {
+    const logLevel = 'something weird';
+    createConfigFile(logLevel).then((configFilePath) => {
+      const configParameters = createConfigParameters(configFilePath);
+      const connection = snowflake.createConnection(configParameters);
+      connection.connect((err) => {
+        if (err) {
+          try {
+            assert.strictEqual(err.message, errorMessages[codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG]);
+            assert.strictEqual(err.code, codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG);
+            done();
+          } catch (e) {
+            done(e);
+          }
+        } else {
+          done(new Error('Error should be thrown'));
+        }
+      });
+    });
+  });
+
+  it('Should apply easy logging config when connection is being opened asynchronously', async function (){
+    // given
+    const logLevel = 'ERROR';
+    const configFilePath = await createConfigFile(logLevel);
+    const configParameters = createConfigParameters(configFilePath);
+    const connection = snowflake.createConnection(configParameters);
+
+    // when
+    await connection.connectAsync();
+
+    // then
+    assert.strictEqual(Logger.getInstance().getLevelTag(), logLevel);
+  });
+
+  it('Should fail to connect asynchronously with wrong easy logging config', async function (){
+    // given
+    const logLevel = 'something weird';
+    const configFilePath = await createConfigFile(logLevel);
+    const configParameters = createConfigParameters(configFilePath);
+    const connection = snowflake.createConnection(configParameters);
+
+    // expect
+    await connection.connectAsync(err => {
+      if (err) {
+        assert.strictEqual(err.message, errorMessages[codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG]);
+        assert.strictEqual(err.code, codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG);
```

Comment:
Code and message are here the same?

Reply:
they are different. Code is codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG.
Message is errorMessages[codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG] so we look into errorMessage array at position of given code.